test/framework/k8s_cluster.go

```diff
@@ -379,16 +380,18 @@ func (c *K8sCluster) DeployKuma(mode string, fs ...DeployOptionsFunc) error {
 		}
 	}
 
-	// wait for the mesh
-	_, err = retry.DoWithRetryE(c.t,
-		"get default mesh",
-		DefaultRetries,
-		DefaultTimeout,
-		func() (s string, err error) {
-			return k8s.RunKubectlAndGetOutputE(c.t, c.GetKubectlOptions(), "get", "mesh", "default")
-		})
-	if err != nil {
-		return err
+	if !opts.skipDefaultMesh {
+		// wait for the mesh
+		_, err = retry.DoWithRetryE(c.t,
```

Comment:
Hmm, waiting for the mesh was somehow a way to figure that the CP is fully operational now. Shall we replace this with another API call?

Reply:
What would be the API call? Related https://github.com/kumahq/kuma/issues/1001

Reply:
Yep, that sounds about right.